### ISSUE TITLE
Calling constructor may break stack map when "assert" option is disabled

### ISSUE THREAD

**Issue by HeartSaVioR** [janino-compiler/janino#119]
This is another one of test failures I've seen in Spark with Janino 3.1.1.

Please add below tests into ExpressionEvaluatorTest and run to see the failures:

```
    @Test public void
    testIssue() throws Exception {
        try {
            // This only happens when assert is disabled, so we disable it temporary.
            CodeContext.class.getClassLoader().setClassAssertionStatus(CodeContext.class.getName(), false);

            IScriptEvaluator eval = new ScriptEvaluator();
            eval.setReturnType(Object.class);
            eval.cook(
                ""
                    + "class A {\n"
                    + "    private int val1;\n"
                    + "    private int val2;\n"
                    + "    public A(int v1, int v2) {\n"
                    + "         val1 = v1;\n"
                    + "         val2 = v2;\n"
                    + "    }\n"
                    + "    public int getValue1() {\n"
                    + "        return val1;\n"
                    + "    }\n"
                    + "    public int getValue2() {\n"
                    + "        return val2;\n"
                    + "    }\n"
                    + "}\n"
                    + "Integer a = 1;"
                    + "Object[] b = new Object[]{\n"
                    // Technically stack map will be broken while calling constructor, but to force checking
                    // stack map and make test fail, we leverage ternary operator.
                    + "a == null ? null : new A(3, 4).getValue1()};\n"
                    + "return b;"
            );
            Object[] ret = (Object[]) eval.evaluate(new Object[]{});
            Assert.assertEquals(3, ret[0]);
        } finally {
            CodeContext.class.getClassLoader().setClassAssertionStatus(CodeContext.class.getName(), false);
        }
    }
```

The culprit looks to be here:

https://github.com/janino-compiler/janino/blob/8dc7efce0325030ed744099dee7d154e8fe742c1/janino/src/main/java/org/codehaus/janino/CodeContext.java#L1439-L1445

Here assertion calls "stateful" method which affects result if assertion is disabled.

**Comment by HeartSaVioR** [comment]
#120

**Comment by aunkrig** [comment]
Merged PR #120 which fixes this issue.
